File: JULY-2025/05-07-2025/p1394.go
func findLucky(arr []int) int {
    
    table := make(map[int]int)
    
    for _ , v := range arr {
        table[v]++
    }

    luckyNum := -1

    for k, v := range table {
        if k == v && k > luckyNum {
            luckyNum = k
        }
    }

    return luckyNum
}

File: JULY-2025/05-07-2025/p1394.erl
find_lucky(Arr) ->
    Counting_array = count_instances(Arr, #{}),
    Luckies = [Num || {Num, Count} <- Counting_array, Num == Count],
    if Luckies == [] -> -1;
        true -> lists:max(Luckies)
    end.

count_instances([], Ret) -> maps:to_list(Ret);
count_instances([A | Arr], Ret) ->
    Counter = maps:get(A, Ret, 0),
    count_instances(Arr, maps:put(A, Counter+1, Ret)).